pkg/virt-controller/services/template.go

```diff
@@ -966,32 +935,6 @@ func (t *templateService) RenderLaunchManifest(vmi *v1.VirtualMachineInstance) (
 		containers = append(containers, sidecar)
 	}
 
-	// XXX: reduce test time. Adding one more container delays the start.
-	// First stdci has issues with that and second we don't want to increase the startup time even more.
-	// At the end the infra container needs to be always there, to allow better default readiness checks.
-	if vmi.Spec.ReadinessProbe != nil {
-		// Infra-ready container
-		readyContainer := k8sv1.Container{
```

Comment:
what was the point of this infra container again? I can't remember, so i'm having trouble reviewing this part. 

I know we're using this container to indicate readiness, but i don't remember why we did this instead of looking at the compute container directly.

Reply:
The `compute` container holds the readiness probe of the VMI if one is specified. Before this PR we had to add an extra container in this case to check if `virt-launcher` was ready before we did the handover.